pages/airdrops/register.tsx

```diff
@@ -145,51 +151,78 @@ const RegisterAirdrop: NextPage = () => {
   }
 
   return (
-    <div className="w-3/4 h-4/4">
+    <div className="relative py-6 px-12 space-y-8">
       <NextSeo title="Register Airdrop" />
-      <h1 className="text-6xl font-bold text-center">Register Airdrop</h1>
-      <div className="my-6">
-        <label className="block mb-2 text-lg font-bold text-gray-300">
-          Airdrop Contract Address
-        </label>
-        <input
-          type="text"
-          className="block p-2.5 w-full text-lg text-black bg-gray-50 rounded-lg border border-gray-300 focus:border-blue-500 dark:border-gray-600 dark:focus:border-blue-500 focus:ring-blue-500 dark:focus:ring-blue-500 dark:placeholder-gray-400"
-          placeholder={
-            contractAddress || 'Please enter your airdrop contract address'
-          }
-          value={contractAddress}
-          onChange={(e) => setContractAddress(e.target.value)}
-        />
+
+      <div className="space-y-8 text-center">
+        <h1 className="text-4xl font-bold">Register Airdrop</h1>
+        <div className="flex justify-center">
+          <AirdropsStepper step={airdrop?.escrow ? 2 : 3} />
+        </div>
+        <p>
+          Now that the contract is deployed, it can be registered to the
+          JunoTools
+        </p>
+      </div>
+
+      <hr className="border-white/20" />
+
+      <div className="space-y-8">
+        <FormControl
+          title="Airdrop contract address"
+          subtitle="Address of the CW20 token that will be airdropped."
+          htmlId="airdrop-cw20"
+        >
+          <Input
+            id="airdrop-cw20"
+            name="cw20"
+            type="text"
+            placeholder="juno1234567890abcdefghijklmnopqrstuvwxyz..."
+            value={contractAddress}
+            onChange={(e) => setContractAddress(e.target.value)}
+          />
+        </FormControl>
+
+        {airdrop && airdrop.escrow && (
```

Comment:
could this be `airdrop?.escrow && {...}` instead? Same for all other conditional logic in this form

Reply:
TypeScript can't comprehend the nullish types somehow. Initial aim is for verbosity, but I'll try changing it to optional chaining.

Reply:
Here's the error if using optional chaining:

<img width="666" alt="image" src="https://user-images.githubusercontent.com/8220954/156430140-fc703298-a242-4893-8281-be4bc4cc5370.png">

Reply:
In this case, since it will only render if that boolean is true, could you do `<JsonPreview title={airdrop!.name} content={airdrop} />`? the syntax may be `airdrop.name!` instead of `airdrop!.name`. Can't remember 100%

Reply:
I forgot typescript can do that 🤦🏻

Reply:
I think ```<JsonPreview title={airdrop!.name} content={airdrop} />``` this broke the page 😄

Reply:
> I think `<JsonPreview title={airdrop!.name} content={airdrop} />` this broke the page 😄

That'd be my bad 😅